src/server/db_slice.cc

```diff
@@ -220,6 +220,24 @@ unsigned PrimeEvictionPolicy::Evict(const PrimeTable::HotspotBuckets& eb, PrimeT
   return 1;
 }
 
+// Helper class to cache and restore fetched_items_ of DbSlice for flows that preempt
+// because some other transaction might conclude and clear the fetched_items_ with OnCbFinish()
+class FetchedItemsRestorer {
+ public:
+  using RestoreType = absl::flat_hash_set<CompactObjectView>;
+  explicit FetchedItemsRestorer(RestoreType* dst) : dst_to_restore_(dst) {
```

Comment:
`fetched_items` are cleraed when `OnCbFinish` gets called which is at the end of a transaction (in fact I think we have a small bug there, I will ping internally). When the fiber `preempts` we should cache this and when it resumes we need to bring it back. The code above provides this guarantee. 

Also if `Find` is called it will be a part of another transaction which will either `conclude` and call `OnCbFinish` or will preempt, cache this value and bring it back when the fiber resumes